### DISCUSSION TITLE
rangeStrategy of pin requires current version to exist on a registry since 37.194.4

### DISCUSSION THREAD

**Discussion by duckmatt** [renovatebot/renovate#28981] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### Please tell us more about your question or problem

Hello,

When we have a `rangeStrategy` of `pin` and have a current version that doesn't exist, we end up with `invalid-value` as the currentVersion and this results in no MR/PR being created.

I believe this behaviour changed since: https://github.com/renovatebot/renovate/pull/27383

I have a minimal example here: https://github.com/duckmatt/renovate-test/

In the action https://github.com/duckmatt/renovate-test/actions/runs/9033730289/job/24824695087 it runs the same code against both renovate versions: 

`37.194.3`:
```
"currentValue": "1.6.0-SNAPSHOT",
...
"updates": [
{
    ...
    "newVersion": "1.12.0",
}],
....
"currentVersion": "1.6.0-SNAPSHOT",
"isSingleVersion": true,
"fixedVersion": "1.6.0-SNAPSHOT"
```

`37.194.4`:
```
"depName": "org.apache.commons:commons-text",
"currentValue": "1.6.0-SNAPSHOT",
...
"skipReason": "invalid-value"
```

-----

In my real scenario the version that doesn't exist is a snapshot on a snapshot specific maven registry, however there are thousands of versions per dep, currently this results in thousands of HTTP calls to check the pom exists when we make the registry available. We are able to get around this by enabling `RENOVATE_EXPERIMENTAL_NO_MAVEN_POM_CHECK` & adding the snapshot registry or using the `rangeStrategy` of `replace`.

Is there any chance of allowing pin to fall through to https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/process/lookup/current.ts#L52-L54 or were we relying previously on a bug?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Please make a _minimal_ reproduction with readme. Your test one is not minimal:

```
 INFO: Dependency extraction complete (repository=duckmatt/renovate-test, baseBranch=main)
       "stats": {
         "managers": {
           "github-actions": {"fileCount": 2, "depCount": 8},
           "gradle": {"fileCount": 2, "depCount": 3},
           "gradle-wrapper": {"fileCount": 1, "depCount": 1}
         },
         "total": {"fileCount": 5, "depCount": 12}
       }
```

I assume only a single dependency is necessary to reproduce this?
